xdsl/pattern_rewriter.py

```diff
@@ -318,8 +318,11 @@ class RewritePattern(ABC):
     A side-effect free rewrite pattern matching on a DAG.
     """
 
+    # The / in the function signature makes the previous arguments positional, see
+    # https://peps.python.org/pep-0570/
+    # This is used by the op_type_rewrite_pattern
     @abstractmethod
-    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter):
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
```

Comment:
It's because `Callable[[A, B, C], None` is a function with three positional parameters, and by default methods have named parameters. The `/` here means that the type of the function returned by the decorator matches the one of being overridden, otherwise the type checker complains.